.github/cloud-samples-tools/pkg/utils/packages.go

```diff
@@ -0,0 +1,49 @@
+/*
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package utils
+
+import (
+	"io/fs"
+	"os"
+	"slices"
+)
+
+// FindAllPackages finds all the packages in the given root directory.
+func FindAllPackages(root string, config Config) ([]string, error) {
```

Comment:
observation: it's curious to me that `FindPackage()` is a method of `Config`, but `FindAllPackages()` is not.

Maybe this is a bad idea, but: would it make sense to create another type, `Package`? Or an aggregate type `Packages`? I know it's late in the review stage for such a large refactor, so feel free to disregard.

Reply:
Yeah, I think it came from an earlier iteration on the prototype, but it actually makes sense to have everything using a config to be a method of config. So I just made that change.